pkg/splunk/controller/configmap.go

```diff
@@ -75,3 +76,50 @@ func GetConfigMapResourceVersion(client splcommon.ControllerClient, namespacedNa
 	}
 	return configMap.ResourceVersion, nil
 }
+
+// GetMCConfigMap gets the MC ConfigMap resource required for that MC
+func GetMCConfigMap(client splcommon.ControllerClient, cr splcommon.MetaObject, namespacedName types.NamespacedName) (*corev1.ConfigMap, error) {
+	var configMap corev1.ConfigMap
+	err := client.Get(context.TODO(), namespacedName, &configMap)
+	if err != nil {
+		//if we don't find mc configmap create and return an empty configmap
+		var configMap corev1.ConfigMap
+		configMap = corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      namespacedName.Name,
+				Namespace: namespacedName.Namespace,
+			},
+			Data: make(map[string]string),
+		}
+		err = splutil.CreateResource(client, &configMap)
+		if err != nil {
+			return nil, err
+		}
+	}
+	err = SetConfigMapOwnerRef(client, cr, namespacedName)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I see these lines are not covered in the UT coverage. I think it can be easily covered by passing a non-existent MC configmap and then SetConfigMapOwnerRef would return error from that.